cdap-app-fabric-tests/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/ScheduledRunTimeTest.java

```diff
@@ -78,7 +79,8 @@ public void after() throws Exception {
   public void testGetNextRun() throws Exception {
     ApplicationId appId = NamespaceId.DEFAULT.app("test");
     deploy(appId, new AppRequest<>(new ArtifactSummary(ARTIFACT_ID.getName(), ARTIFACT_ID.getVersion().getVersion())));
-
+    JsonObject result = getAppDetails(NamespaceId.DEFAULT.getNamespace(), "test");
+    appId = new ApplicationId(appId.getNamespace(), appId.getApplication(), result.get("appVersion").getAsString());
```

Comment:
Does this mean that for anyone who relies for their automation to work with "-SNAPSHOT", it would break?

Reply:
if a user wants to retrieve their -SNAPSHOT version they can use the "versioned" endpoint (such as https://github.com/cdapio/cdap/blob/develop/cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java#L374)  but yes, they cannot create an app with a custom version such as -SNAPSHOT. 

The details are in the design doc

Reply:
It's not about deploy. As discussed, this test should work "as is". It will call `apps/%s/versions/%s/schedules/%s/enable` with -SNAPSHOT as parameter and system should map "-SNAPSHOT" to latest. Please revert test changes or if you want to test both, add another test

Reply:
so here -SNAPSHOT is not the latest version. We called deploy at L81 which creates a new version that is the latest and as such the schedules operate on the latest version - we do not use the versioned api in this test

Reply:
discussed offline - will ignore this test